gpt4all-chat/modellist.cpp

```diff
@@ -1202,7 +1202,7 @@ void ModelList::updateModelsFromDirectory()
                             { InstalledRole, true },
                             { FilenameRole, filename },
                             // FIXME: WE should change this to use a consistent filename for online models
-                            { OnlineRole, filename.startsWith("chatgpt-") || filename.startsWith("nomic-") },
```

Comment:
Yeah, you might as well erase the nomic condition here. The point is we have to decide if ending in '.txt' is enough or if we want to have a prefix or some other conditional naming heuristic. What do you guys think? I think I'd prefer a standard prefix... indicating an online model?

Reply:
I don't know, we know offline models will never be text file, so I'm fine with checking if the file ends with '.txt'

Reply:
Yeah, but it also means that if the user happens to change the model directory to a place with a bunch of txt files in it we're going to be interpreting those files as models... that doesn't sound like a good idea hence the need for a naming convention.

Reply:
you're right but even with prefixes there still could be conflicts. I think we either need another extension, or store the api data separately the same way you do with other models, and check for specific files

Reply:
The only thing the file contains is the api key. I don't want to store this in a setting as that has a greater possibility of going 'stale' see the discussion on the PR for cleaning up stale settings.

Reply:
For as long as we are encouraging users to add/remove models directly in the models folder and giving them the option of changing its path whenever they want, I think it would be simplest to continue to store the API key in a file.

Semantically though, these are not plain text files, which conventionally contain human-readable text - they are data files with a strict format. So unless we want to keep a strict whitelist of filenames (which at this point isn't a big deal), we should either store them in a structured data format like JSON (e.g. `gpt4all-openai-mistral.json` - this format could be extended to also contain the URL for the API endpoint) or use a custom file extension (e.g. `gpt4all-mistral.rmodel`, where rmodel stands for "remote model").

I think having "gpt4all" in the filename would make the source of the file clearer in case the user shares the model directory between multiple local LLM apps.

Reply:
Ok, let's go with this:

* a prefix of 'gpt4all-' to the file name
* a file extension of '.rmodel'
* either JSON or INI for the file format (up to the implementer to decide which)
* the model name is stored in the file
* the api key is stored in the file
* the url is stored in the file

Reply:
will this file be stored in the chat metadata along the other models ? Or does it replace the current .txt file ? Because in the latter case, it doesn't solve the issue of storing the model name, to be able to copy it to that file.